### PR TITLE
Add test coverage

### PR DESCRIPTION
**Pull request by acuppy**
Run the suite: `bundle exec rspec spec`

### CODE REVIEW

File: daemons.gemspec [hunk 1]
```diff
@@ -23,4 +23,8 @@ Gem::Specification.new do |s|
   EOF
 
   s.files = `git ls-files README.md LICENSE Releases lib examples`.split
+
+  s.add_development_dependency 'rspec', '>= 3.1'
```
Comment (sodabrew) [inline daemons.gemspec]: rspec tends to change syntax over time - I would advise a squiggly requirement like  `~> 3.1` (or a combined requirement, like `'rspec', '>= 3.1', '~> 3.1'`).

**Comment by sodabrew** [comment]
This is totally awesome. Still in progress?

**Comment by acuppy** [comment]
Yes.  I haven't had a lot of time.  However, if it's passing, it is better than nothing to have a test suite in place.

File: daemons.gemspec (hunk 1 above)
Comment (acuppy) [inline daemons.gemspec]: That's fair

**Comment by acuppy** [comment]
@thuehlinger this is a good time to merge this PR.  It's not complete by any means, but it's passing and establishes some core contribution guidelines.

**Comment by sodabrew** [comment]
Rather than cherry-picking commits from master onto the branch, I'd recommend to rebase the branch up to master.

**Comment by acuppy** [comment]
@sodabrew it was rebased against `master` but not pushed.  It's up to date now.

**Comment by acuppy** [comment]
Let me know if there's any other adjustments to get in there.  Thanks!

**Comment by pboling** [comment]
:+1: This gem is important, and a test suite is also important for many corporate FOSS software approval processes.

**Comment by acuppy** [comment]
@thuehlinger any word on merging this in?

**Comment by pboling** [comment]
@acuppy The project may need additional maintainers that frequent Github more often.

**Comment by thuehlinger** [comment]
Merged. Thanks for the great work and my apologies for the delay.

**Comment by acuppy** [comment]
All good :)

**Pull request merged**